modules/saml/www/sp/metadata.php

```diff
@@ -58,6 +58,7 @@
 
 $index = 0;
 $eps = array();
+$supported_protocols = array();
```

Comment:
I think this is the way to go and I really want to see this merged. However, there's a problem with backwards-compatibility. Our rule of thumb is that we try to keep backwards-compatibility always between minor versions (1.14 in this case). Metadata is a specially critical case where backwards-compatibility is essential for interoperability, so we always try to be extra careful with it.

In this case, this change would mean different metadata, so I think we need a configuration option to enable this, and set it to keep the current behaviour by default. Then you can switch that option to remove unused protocols depending on your configuration.

Later on, when we release 2.0, I'd get rid of that option and make this the default behaviour. I wouldn't spend much time on documenting the configuration option therefore.

How does that sound?